src/compiler/checker.ts

```diff
@@ -8831,10 +8834,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     return setTextRange(context, setEmitFlags(name, EmitFlags.NoAsciiEscaping), node);
                 }
                 const updated = visitEachChildWorker(node, c => attachSymbolToLeftmostIdentifier(c), /*context*/ undefined);
-                if (updated !== node) {
-                    setTextRange(context, updated, node);
-                }
-                return updated;
+                return setTextRange(context, updated, node);

```

Comment:
The containing function was only cloning the leftmost identifier in a qualified name, leaving the other ones intact - but they could also be from a different source file than the target file. So when the leftmost one gets updated, the right ones need to be (potentially) cloned too.

But even the node path containing the leftmost identifier wasn't handled quite right here because that the return value of `setTextRange` was ignored.